MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js

```diff
@@ -101,13 +109,56 @@ define([
          * Verify directory filter on init event, select folder per directory filter state
          */
         checkChipFiltersState: function () {
```

Comment:
I think we can simplify this method, we are doing 3 times pretty much the same thing:
```
if (this.isFolderExistsInTree(currentFilterPath)) {
                    selectedId === currentFilterPath  || this.locateNode(currentFilterPath);
                } else {
                    this.selectStorageRoot();
                }
```

I believe something like this should work:
```
        /**
         * Verify directory filter on init event, select folder per directory filter state
         */
        checkChipFiltersState: function () {
            var currentFilterPath = this.filterChips().filters.path,
                isMediaBrowser = !_.isUndefined(window.MediabrowserUtility),
                currentTreePath;

            currentTreePath = (this.isFiltersApplied(currentFilterPath) || !isMediaBrowser) ? currentFilterPath :
                Base64.idDecode(window.MediabrowserUtility.pathId) ;

            if (this.isFolderExistsInTree(currentTreePath)) {
                this.locateNode(currentTreePath);
            } else {
                this.selectStorageRoot();
            }
        },
```

We just need to change the locateNode, to perform the validation as shown below:

```
        /**
         * Locate and higlight node in jstree by path id.
         *
         * @param {String} path
         */
        locateNode: function (path) {
            var selectedId =  $(this.directoryTreeSelector).jstree('get_selected').attr('id');

            if (path === selectedId) {
                return;
            }
            path = path.replace(/\//g, '\\/');
            $(this.directoryTreeSelector).jstree('open_node', '#' + path);
            $(this.directoryTreeSelector).jstree('select_node', '#' + path, true);

        }
```